packages/asset-swapper/src/utils/market_operation_utils/types.ts

```diff
@@ -49,13 +45,28 @@ export interface NativeFillData extends FillData {
     order: SignedOrderWithFillableAmounts;
 }
 
+export interface CurveFillData extends FillData {
+    poolAddress: string;
+    fromTokenIdx: number;
+    toTokenIdx: number;
+}
+
+export interface BalancerFillData extends FillData {
+    poolAddress: string;
+}
+
+export interface UniswapV2FillData extends FillData {
+    tokenAddressPath: string[];
+}
+
 /**
  * Represents an individual DEX sample from the sampler contract.
  */
-export interface DexSample {
+export interface DexSample<TFillData extends FillData = FillData> {
```

Comment:
I feel like having the type parameter default here is nice, since most sources (Uniswap, Eth2Dai, Kyber, LiquidityProvider, MultiBridge) aren't using the optional `fillData` field for now. Saves some `<Filldata>`s